sdk/examples/builder_sample.rs

```diff
@@ -0,0 +1,169 @@
+    json!({
+        "title": title,
+        "format": format,
+        "claim_generator_info": [
+            {
+                "name": "c2pa test",
+                "version": env!("CARGO_PKG_VERSION")
+            }
+        ]
+    })
+    .to_string()
+}
+
+/// Capture an ingredient from a stream and return the resulting manifest store as a Vec<u8>
+///
+/// This can be constructed using existing methods with a few tricks
+fn capture_ingredient<R>(format: &str, stream: &mut R) -> Result<Vec<u8>>
+where
+    R: Read + Seek + Send,
+{
+    let mut builder = Builder::new();
+
+    // we need to manually add the ingredient stream since it has a different format than the output .c2pa
+    builder.add_ingredient_from_stream(
+        json!({
+            "title": "Archived Ingredient",
+            "relationship": "parentOf",
+            "label": "test_ingredient"
+        })
+        .to_string(),
+        format,
+        stream,
+    )?;
+    // add the opened action to ensure this is a valid manifest
+    builder.add_action(json!(
+        {
+            "action": "c2pa.opened",
+            "parameters": {
+                "ingredientIds": ["test_ingredient"],
+            }
+        }
+    ))?;
+
+    // sign a c2pa only manifest store by using a null input stream and application/c2pa as the format.
+    let mut null_stream = Cursor::new([]);
+    let mut output = Cursor::new(Vec::new());
+    builder.sign(
+        &Settings::signer()?,
+        "application/c2pa",
+        &mut null_stream,
+        &mut output,
+    )?;
+
+    Ok(output.into_inner())
+}
+
+fn main() -> Result<()> {
+    let format = "image/jpeg";
+    let ingredient_name = "Restored Ingredient";
+    let mut ingredient_source = Cursor::new(INGREDIENT_IMAGE);
+
+    Settings::from_toml(include_str!("../tests/fixtures/test_settings.toml"))?;
+
+    // Here we capture an ingredient with its validation into a c2pa_data object.
+    let ingredient_c2pa = capture_ingredient(format, &mut ingredient_source)?;
+    // The ingredient_c2pa can be saved to a file, blob storage, a database, or wherever you want to keep it.
+    // For this example we will just keep it in memory and add it to a new manifest
+
+    // Now create a new builder and set the intent to create a new manifest store
+    // We will add the ingredient as a componentOf relationship
+    let mut builder = Builder::from_json(&manifest_def("Builder Sample", format))?;
+    builder.set_intent(c2pa::BuilderIntent::Create(DigitalSourceType::Empty));
+
+    // Now add our saved ingredient as a c2pa stream.
+    // When the format is "application/c2pa" we will fetch the parent stream from the manifest.
+    builder.add_ingredient_from_stream(
+        json!({   // we can override any values in the saved ingredient here
+            "title": ingredient_name,
+            "relationship": "componentOf",// This relationship will override the one in the saved ingredient
+            "label": "component_image"
```

Comment:
Maybe we should put "component_image" in a variable and add a little bit of documentation about how this connects ingredients with actions.